src/textual/dom.py

```diff
@@ -1206,12 +1206,11 @@ def ancestors_with_self(self) -> list[DOMNode]:
         Returns:
             A list of nodes.
         """
-        nodes: list[MessagePump | None] = [self]
-        add_node = nodes.append
-        node: MessagePump | None = self
-        while (node := node._parent) is not None:
-            add_node(node)
-        return cast("list[DOMNode]", nodes)
+        nodes = [self]
```

Comment:
This change will make this method slower. If it is to satisfy Mypy, can we not have both fast and correct?

Reply:
the add_node = node.append trick is actually slower now

Reply:
I've looked into this and have benchmarked some options:

(there's not all that much in it, even if you really crank up the number of nodes)

https://gist.github.com/graingert/23d43b7fb7c8d5663cc355e8390f19d9

the option I want to go for is:

```python
    nodes = []
    node: DOMNode | None = self

    while True:
        if node is None:
            return nodes

        nodes.append(node)
        node = node._parent
```

some remarks:  repeatedly looking up .append is actually much faster, doing the condition in the body of a while loop is faster this is probably a bug!